### PR TITLE
Fixes #735: Add repo-sync script for charts

### LINKED ISSUE #590: Publish charts/catalog helm chart to a public chart repository
A good place to publish would be a public Google Cloud Storage or AWS S3 bucket, as various helm charts do.

See https://github.com/kubernetes/helm/blob/master/docs/charts.md#chart-repositories for a description on what a chart repository is.

### LINKED ISSUE #735: Publish helm charts to a helm repo
This is a follow up to https://github.com/kubernetes-incubator/service-catalog/pull/728#discussion_r112563258

Currently, we reference version-tagged images in our helm charts, not `canary`. This keeps our walk-through stable, but long-term I'd like to see us treat our helm charts as published artifacts just like our docker images.

This issue is for the following:

1. Publish helm charts to a helm repo, just like we publish images to quay (iirc, quay now can host helm charts - I don't know any more details right now)
2. Change the image tags in our helm charts back to canary
3. Modify the walk-through to point to the versioned helm charts in the helm repo

### PR DESCRIPTION
**Pull request by viglesiasce**
#735

This adds the repo-sync script that will allow the service catalog charts to be uploaded to a Helm repository.

Currently the steps from a user perspective would be:
```shell
$ helm repo add svc-catalog https://svc-catalog-charts.storage.googleapis.com
$ helm install svc-catalog/ups-broker
$ helm install svc-catalog/catalog
```

I have a CI job set up that will only run on when tags are pushed. We can set that up however you'd like though. Once this is merged I will point the CI job at this repo rather than my fork.

### CODE REVIEW

File: test/repo-sync.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash -xe
+# Copyright 2016 The Kubernetes Authors All rights reserved.
```
Comment (vaikas-google) [inline test/repo-sync.sh]: Not sure if the boilerplate is wrong because the year is wrong, or maybe a space between lines 1&2 is necessary?

**Comment by vaikas-google** [comment]
Travis is frowning upon the boilerplate:
Running repo-infra verify scripts
Boilerplate header is wrong for: ././test/repo-sync.sh
make: *** [verify] Error 1

**Comment by vaikas-google** [comment]
And thanks for doing this!!

**Comment by pmorie** [comment]
@viglesiasce can you fix the boilerplate header quick?

**Comment by viglesiasce** [comment]
@pmorie @vaikas-google should be all set!

**Comment by pmorie** [comment]
@viglesiasce LGTM, can you provide some info abiunt the CI job that does this? Do we need to do anything?

**Comment by viglesiasce** [comment]
Shouldn't need to do anything its running on the same CI server as the main
charts repo sync. I just cloned the job. Once this PR merges, I will flip
it to watch this repo instead of my fork. ATM it is only watching for Git
tags of this repo.

Can/should I add this info somewhere or should we wait a bit until we have
this syncing to the main charts repo?

**Comment by arschles** [comment]
@viglesiasce can you add something to a doc in this repo to indicate where things are publishing to? not sure if this is already done. if so, forgive my tardiness & redundancy

**Comment by MHBauer** [comment]
@viglesiasce can you help us investigate why we haven't seen any update after the v0.1.9 release?

**Comment by viglesiasce** [comment]
Yep. Will take a look. I believe this is related to a failure last week in
the main Charts sync.

**Comment by viglesiasce** [comment]
Hey Folks!!

I got things back working but only the 0.12.0 was spit out. Things should auto build moving forward. That said, we are looking at better ways to handle this simple automation in a more public/visible way. Will keep you posted.

**Comment by MHBauer** [comment]
@viglesiasce is there a charts status page?

**Pull request merged**